### PR TITLE
Extend tests coverage

### PR DESCRIPTION
**Pull request by IvanNardi**
Now there is at least one flow under `tests/pcap` for 249 protocols out
of the 284 ones supported by nDPI.

The 35 protocols without any tests are:

* P2P/sharing protocols: DIRECT_DOWNLOAD_LINK, OPENFT, FASTTRACK,
EDONKEY, SOPCAST, THUNDER, APPLEJUICE, DIRECTCONNECT, STEALTHNET

* games: CSGO, HALFLIFE2, ARMAGETRON, CROSSFIRE, DOFUS, FIESTA,
FLORENSIA, GUILDWARS, MAPLESTORY, WORLD_OF_KUNG_FU

* voip/streaming: VHUA, ICECAST, SHOUTCAST, TVUPLAYER, TRUPHONE

* other: AYIYA, SOAP, TARGUS_GETDATA, RPC, ZMQ, REDIS, VMWARE, NOE,
LOTUS_NOTES, EGP, SAP

Most of these protocols (expecially the P2P and games ones) have been
inherited by OpenDPI and have not been updated since then: even if they
are still used, the detection rules might be outdated.

However code coverage (of `lib/protocols`) only increases from 65.6% to
68.9%.

Improve Citrix, Corba, Fix, Aimini, Megaco, PPStream and Some/IP
dissection.
Treat IPP as a HTTP sub protocol.
Fix Cassandra false positives.

Remove `NDPI_PROTOCOL_QQLIVE` and `NDPI_PROTOCOL_REMOTE_SCAN`:
these protocol ids are defined but they are never used.

Remove Collectd support: its code has never been called. If someone is
really interested in this protocol, we can re-add it later, updating the
dissector.

Add decoding of PPI (Per-Packet Information) data link type.

### CODE REVIEW

File: tests/result/socks-http-example.pcap.out [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+DPI Packets (TCP):	27	(9.00 pkts/flow)
+Confidence Match by port    : 1 (flows)
+Confidence DPI              : 2 (flows)
+
+SOCKS	31	5603	2
+Cassandra	15	2780	1
+
+	1	TCP 10.180.156.185:53535 <-> 10.180.156.249:1080 [proto: 172/SOCKS][ClearText][Confidence: Match by port][cat: Web/5][10 pkts/832 bytes <-> 7 pkts/2073 bytes][Goodput ratio: 19/77][0.01 sec][bytes ratio: -0.427 (Download)][IAT c2s/s2c min/avg/max/stddev: 0/0 1/1 4/3 2/1][Pkt Len c2s/s2c min/avg/max/stddev: 66/66 83/296 212/1514 43/500][PLAIN TEXT (uGET / HTTP/1.1)][Plen Bins: 57,0,0,0,28,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,14,0,0]
+	2	TCP 10.180.156.185:53534 <-> 10.180.156.249:1080 [proto: 264/Cassandra][ClearText][Confidence: DPI][cat: Database/11][8 pkts/711 bytes <-> 7 pkts/2069 bytes][Goodput ratio: 24/77][0.05 sec][bytes ratio: -0.488 (Download)][IAT c2s/s2c min/avg/max/stddev: 0/0 8/12 47/46 18/20][Pkt Len c2s/s2c min/avg/max/stddev: 66/66 89/296 212/1514 47/500][Risk: ** Known Protocol on Non Standard Port **][Risk Score: 50][PLAIN TEXT (GET / HTTP/1.1)][Plen Bins: 40,0,0,0,40,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,20,0,0]
```
Comment (utoni) [inline tests/result/socks-http-example.pcap.out]: Shouldn't that SOCKS as well?

**Review by utoni — APPROVED** [review]
Thanks for cleaning up some protocol mess.
RiP collectd. I will probably reimplement that protocol later. This time with a pcap.
The coverage for protocol dissectors did only slightly increase, but there is some progress. At some point we need also think about using a decent test framework for (nDPI core) unit tests e.g. criterion. But that is another story for another time.

File: tests/result/socks-http-example.pcap.out (hunk 1 above)
Comment (IvanNardi) [reply tests/result/socks-http-example.pcap.out]: It should.. I'll take a look

File: tests/result/lisp_registration.pcap.out [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	2	UDP 10.0.123.2:4342 <-> 10.0.123.1:4342 [proto: 236/LISP][ClearText][Confidence: DPI][cat: Cloud/13][4 pkts/568 bytes <-> 4 pkts/568 bytes][Goodput ratio: 70/70][0.01 sec][bytes ratio: 0.000 (Mixed)][IAT c2s/s2c min/avg/max/stddev: 0/3 0/1 0/3 0/1][Pkt Len c2s/s2c min/avg/max/stddev: 130/130 142/142 154/154 8/8][Plen Bins: 0,0,25,75,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+
+
+Undetected flows:
+	1	TCP 10.0.123.3:52995 <-> 10.0.123.1:4342 [proto: 0/Unknown][ClearText][Confidence: Unknown][4 pkts/894 bytes <-> 3 pkts/715 bytes][Goodput ratio: 74/76][0.36 sec][bytes ratio: 0.111 (Mixed)][IAT c2s/s2c min/avg/max/stddev: 1/1 120/73 213/145 88/72][Pkt Len c2s/s2c min/avg/max/stddev: 60/60 224/238 714/586 283/246][Plen Bins: 33,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,33,0,0,0,33,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
```
Comment (utoni) [inline tests/result/lisp_registration.pcap.out]: Those flows may be related to LISP as well?

File: tests/result/snmp.pcap.out [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	9	UDP 65.2.162.193:59988 <-> 130.70.149.185:161 [proto: 14/SNMP][ClearText][Confidence: DPI][cat: Network/14][2 pkts/170 bytes <-> 2 pkts/196 bytes][Goodput ratio: 50/57][0.26 sec][PLAIN TEXT (public)][Plen Bins: 0,100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+	10	UDP 176.211.60.43:43015 <-> 97.0.115.163:161 [proto: 14/SNMP][ClearText][Confidence: DPI][cat: Network/14][2 pkts/170 bytes <-> 2 pkts/196 bytes][Goodput ratio: 50/57][0.26 sec][PLAIN TEXT (public)][Plen Bins: 0,100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+
+
+Undetected flows:
+	1	UDP 10.99.8.88:43242 <-> 10.100.253.146:161 [VLAN: 1308][proto: 0/Unknown][ClearText][Confidence: Unknown][3 pkts/2367 bytes <-> 3 pkts/1502 bytes][Goodput ratio: 94/91][0.11 sec][bytes ratio: 0.224 (Upload)][IAT c2s/s2c min/avg/max/stddev: 21/18 44/20 67/21 23/2][Pkt Len c2s/s2c min/avg/max/stddev: 611/75 789/501 1143/717 250/301][PLAIN TEXT (public)][Plen Bins: 16,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,33,0,0,33,0,0,0,0,0,0,0,0,0,0,0,0,0,16,0,0,0,0,0,0,0,0,0,0,0,0,0]
```
Comment (utoni) [inline tests/result/snmp.pcap.out]: No SNMP?

File: tests/result/lisp_registration.pcap.out (hunk 2 above)
Comment (IvanNardi) [reply tests/result/lisp_registration.pcap.out]: Another think I'll take a look into...

File: tests/result/socks-http-example.pcap.out (hunk 1 above)
Comment (IvanNardi) [reply tests/result/socks-http-example.pcap.out]: Fixed

File: tests/result/lisp_registration.pcap.out (hunk 2 above)
Comment (IvanNardi) [reply tests/result/lisp_registration.pcap.out]: Fixed in last push

File: tests/result/snmp.pcap.out (hunk 3 above)
Comment (IvanNardi) [reply tests/result/snmp.pcap.out]: Fixed

**Review by utoni — APPROVED** [review]

**Pull request merged**